### PR TITLE
Add Support for Creating Contacts with Optional Company Information

### LINKED ISSUE #3: See all contacts FE
A logged in user should be able to see a list of all their own contacts

**I couldn't figure out how to remove the checkboxes from my mock up 🙃. Those are not needed at this time.**

Image

Checklist:
- [x] Create Contacts component
- [x] Fetch request to the BE for the users contacts
- [x] Tailwind CSS styling to match the mockup
- [ ] Fetch request to grab all companies for a user to populate the table 

Blockers:
BE Contacts endpoint needs to be completed to make the fetch requests. I am my own blocker!
Learning how to style with Tailwind and getting a better understanding of TypeScript.

### LINKED ISSUE #60: BE Fix Contact Serializer and Nested contact route for create action
Refactored the contact serializer to add company info if we have a company name when creating a contact. 
Added tests for the contacts controller new business logic for the new nested contact routes.
Added missing nested contact route for create action.

Refactoring of ContactsController to make skinny has been moved to ticket #58

### PR DESCRIPTION
**Pull request by CCirbo**
## Type of Change
- [x] feature ⛲
- [x] testing 🧪

## Description
•	Refactored ContactSerializer to include company information if a contact is created with a company.
•	Added tests for the ContactController to validate logic for handling nested contact routes and companies.
•	Implemented a create action for the nested contacts route.

## Motivation and Context
These changes allow a contact to be created with a company.name/company_id or without a company, ensuring flexibility and maintaining a consistent user experience. The additional tests ensure 100% code coverage in SimpleCov for the Contact Controller's new logic.
closes #60 
closes #3 

## Related Tickets
•	Refactoring of the ContactsController has been moved to ticket #58.

## Added Test?
- [x] Yes 🫡
 Request/Controller Specs?
- [ ] No 🙅
- [x] All previous tests still pass 🥳

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md
Comment (Sgalvin36) [inline README.md:654]: Perhaps adding some clarification on which of the fields are absolutely required. I saw in the tests you were using a minimal params when creating. So while it is very helpful to have the full JSON as an example, having some information on which fields will trigger the 404 error if not included can clear that up. For example, first and last name are required while email is an optional field.

File: app/controllers/api/v1/contacts_controller.rb
Comment (Sgalvin36) [inline app/controllers/api/v1/contacts_controller.rb:31]: Similar to the company controller, this could potentially be refactored into the model. While not necessary for this specific PR, something to make note of, especially if there is any more logic added to the action at a later time.

File: config/routes.rb
Comment (jimmacur) [inline config/routes.rb:1]: Can you let me know why we need a contacts :create under both contacts and /companies/contacts?

**Review by Sgalvin36 — CHANGES_REQUESTED** [review]
The changes made to the controllers looks great. I appreciate the checks for companies to make those connections while the contact is being made. 
I asked for some more information to be provided in the README, I don't see anything in the code specifically that needs to be addressed before approval.

File: spec/requests/api/v1/contacts/create_spec.rb
Comment (jimmacur) [inline spec/requests/api/v1/contacts/create_spec.rb:1]: Extremely thorough testing!

**Review by jimmacur — APPROVED** [review]
Things are looking very good here. Ready to merge!

File: app/controllers/api/v1/contacts_controller.rb
Comment (CCirbo) [reply app/controllers/api/v1/contacts_controller.rb:31]: Yes, definitely. There is a ticket up for that, #58 in backlog.

File: config/routes.rb
Comment (CCirbo) [reply config/routes.rb:1]: Great question!  In Create a Contact, we need that companies/create to create a contact with a company. If we create a contact without a company, then it goes to the contacts/create route. It was the last piece needed for that page.

File: README.md
Comment (reneemes) [reply README.md:654]: Thank you for pointing this out. Due to how the DB is currently set up only the first name, last name, and company email, depending on which URL being used. I will update the README right now to make note of when these fields are required.

File: app/controllers/api/v1/contacts_controller.rb
Comment (Sgalvin36) [reply app/controllers/api/v1/contacts_controller.rb:31]: Ope! I'm sorry I forgot that you mentioned that in the stand up.

File: README.md
Comment (reneemes) [reply README.md:656]: The Read Me has been updated.

**Review by Sgalvin36 — APPROVED** [review]
Thank you for adding the clarification in the README

**Pull request merged**
